util/crypto/src/secp/generator.rs

```diff
@@ -13,16 +13,23 @@ impl Generator {
     }
 
     pub fn random_privkey(&self) -> Privkey {
-        let mut rng = rand::thread_rng();
-        key::SecretKey::new(&SECP256K1, &mut rng).into()
+        let mut random_slice: [u8; secp256k1::constants::SECRET_KEY_SIZE] = rand::random();
+        loop {
+            match SecretKey::from_slice(&random_slice) {
+                Ok(sec) => return sec.into(),
+                Err(_) => {
+                    random_slice = rand::random();
+                }
+            }
+        }
     }
 
     pub fn random_keypair(self) -> Result<(Privkey, Pubkey), Error> {
```

Comment:
Can we return turple here? since we have changed the inner logic of `random_privkey` which always return a valid key.

Reply:
It really works, but since I have to modify all the code that uses this API, I am hesitant.

Reply:
`random_keypair ` is only used in testing code if I remember correctly.

Reply:
Oh，let‘s do it right now